paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt

```diff
@@ -100,31 +100,25 @@ abstract class PrepareResourcesTask : DefaultTask() {
       mainPackage
     }
 
-    out.bufferedWriter()
-      .use {
-        it.write(mainPackage)
-        it.newLine()
-        it.write(mergeResourcesOutputDir.get())
-        it.newLine()
-        it.write(targetSdkVersion.get())
-        it.newLine()
-        // Use compileSdkVersion for system framework resources.
-        it.write("platforms/android-${compileSdkVersion.get()}/")
-        it.newLine()
-        it.write(mergeAssetsOutputDir.get())
-        it.newLine()
-        it.write(resourcePackageNames)
-        it.newLine()
-        it.write(projectResourceDirs.joinFiles(projectDirectory))
-        it.newLine()
-        it.write(moduleResourceDirs.joinFiles(projectDirectory))
-        it.newLine()
-        it.write(aarExplodedDirs.joinFiles(gradleUserHomeDirectory))
-        it.newLine()
-        it.write(projectAssetDirs.joinFiles(projectDirectory))
-        it.newLine()
-        it.write(aarAssetDirs.joinFiles(gradleUserHomeDirectory))
-        it.newLine()
+    val properties = buildMap<String, String> {
+      this["mainPackage"] = mainPackage
+      this["mergeResourcesOutputDir"] = mergeResourcesOutputDir.get()
+      this["targetSdkVersion"] = targetSdkVersion.get()
+      // Use compileSdkVersion for system framework resources.
+      this["platformDir"] = "platforms/android-${compileSdkVersion.get()}/"
+      this["mergeAssetsOutputDir"] = mergeAssetsOutputDir.get()
+      this["resourcePackageNames"] = resourcePackageNames
+      this["projectResourceDirs"] = projectResourceDirs.joinFiles(projectDirectory)
+      this["moduleResourceDirs"] = moduleResourceDirs.joinFiles(projectDirectory)
+      this["aarExplodedDirs"] = aarExplodedDirs.joinFiles(gradleUserHomeDirectory)
+      this["projectAssetDirs"] = projectAssetDirs.joinFiles(projectDirectory)
+      this["aarAssetDirs"] = aarAssetDirs.joinFiles(gradleUserHomeDirectory)
+    }
+    out.bufferedWriter().use { writer ->
+      properties.forEach { (k, v) ->
+        writer.write("$k=$v")
```

Comment:
Why don't you use an actual Properties object for serialisation/deserialisation. Yes, it's old, but it takes care of escaping and is standard.

One benefit would be having multiline values, meaning we don't need to join and split commas on both sides, which would make debugging the file easier.

Reply:
Considered it, but discovered when writing Properties, it adds a timestamp comment to the beginning of the file contents.  Could've worked around that, but figured it's just as easy to do this.

Re: escaping, I don't think any of these config values require that?

Re: multiline values, that's fair, although I've never been a fan of the multiline syntax:
```
multi.prop=value1
multi.prop=value2
...
```

Perhaps migrating to a JSON format would be a better idea?

Reply:
> it adds a timestamp comment to the beginning of the file

https://docs.gradle.org/current/dsl/org.gradle.api.tasks.WriteProperties.html

> I don't think any of these config values require that?

They're file names (user input), you never know.

> I've never been a fan of the multiline syntax:

TIL, but I was thinking more like (without having to worry about how it is generated/read):
```
multi.prop=value1\
value2
```

> join/split being annoying, perhaps migrating to a JSON format

Potentially, you could use the same data class on both sides. But then you need a dependency on some serialization lib.

Reply:
going with json!